src/seedu/addressbook/data/person/Address.java

```diff
@@ -37,19 +77,51 @@ public static boolean isValidAddress(String test) {
 
     @Override
     public String toString() {
-        return value;
+        
+        ArrayList<String> addressComponents = new ArrayList<>();
+        
+        if (block != null) {
+            addressComponents.add(block.toString());
+        }
+        
+        if (street != null) {
+            addressComponents.add(street.toString());
+        }
+        
+        if (unit != null) {
+            addressComponents.add(unit.toString());
+        }
+        
+        if (postalCode != null) {
+            addressComponents.add(postalCode.toString());
+        }
+        
+        StringBuilder addressValue = new StringBuilder();
+        String divider = ", ";
+        
+        for (int i = 0; i < addressComponents.size(); i++){
+            addressValue.append(addressComponents.get(i));
+            if (i != addressComponents.size() - 1) {
+                addressValue.append(divider);
+            }
+        }
+        
+        return addressValue.toString();
     }
 
     @Override
     public boolean equals(Object other) {
         return other == this // short circuit if same object
                 || (other instanceof Address // instanceof handles nulls
-                && this.value.equals(((Address) other).value)); // state check
+                && Objects.equals(this.block, ((Address) other).block)
+                    && Objects.equals(this.street, ((Address) other).street)
+                            && Objects.equals(this.street, ((Address) other).street)
+                                    && Objects.equals(this.postalCode, ((Address) other).postalCode)); // state every single address component.
```

Comment:
The staircase was auto-generated by Eclipse and I thought it was meant to be as such.